features/dd-sdk-android-trace/src/test/kotlin/com/datadog/android/trace/opentelemetry/OtelContextTest.kt

```diff
@@ -0,0 +1,204 @@
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import io.opentelemetry.api.trace.Span
+import io.opentelemetry.context.Context
+import io.opentelemetry.context.ContextKey
+import io.opentelemetry.context.Scope
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class OtelContextTest {
+
+    @Mock
+    private lateinit var mockContext: Context
+
+    @Mock
+    private lateinit var mockCurrentSpan: Span
+
+    @Mock
+    private lateinit var mockRootSpan: Span
+
+    private lateinit var testedContext: OtelContext
+
+    @BeforeEach
+    fun setUp() {
+        mockContext = mock()
+        testedContext = OtelContext(mockContext, mockCurrentSpan, mockRootSpan)
+    }
+
+    // region get by key
+
+    @Test
+    fun `M return current span W get { key is OTEL_CONTEXT_SPAN_KEY }`() {
+        // Given
+        val key = ContextKey.named<Span>(OtelContext.OTEL_CONTEXT_SPAN_KEY)
+
+        // When
+        val result = testedContext.get(key)
+
+        // Then
+        assertThat(result).isEqualTo(mockCurrentSpan)
+    }
+
+    @Test
+    fun `M return root span W get { key is OTEL_CONTEXT_ROOT_SPAN_KEY }`() {
+        // Given
+        val key = ContextKey.named<Span>(OtelContext.OTEL_CONTEXT_ROOT_SPAN_KEY)
+
+        // When
+        val result = testedContext.get(key)
+
+        // Then
+        assertThat(result).isEqualTo(mockRootSpan)
+    }
+
+    @Test
+    fun `M delegate to wrapped Context W get { key is not recognized }`(forge: Forge) {
+        // Given
+        val fakeKeyName = forge.anAlphabeticalString()
+        val key = ContextKey.named<String>(fakeKeyName)
+        val fakeKeyValue = forge.anAlphabeticalString()
+        whenever(mockContext.get(key)).thenReturn(fakeKeyValue)
+
+        // When
+        val result = testedContext.get(key)
+
+        // Then
+        assertThat(result).isEqualTo(fakeKeyValue)
+    }
+
+    // endregion
+
+    // region with key and value
+
+    @Test
+    fun `M return OtelContext W with { key is OTEL_CONTEXT_SPAN_KEY }`() {
+        // Given
+        val key = ContextKey.named<Span>(OtelContext.OTEL_CONTEXT_SPAN_KEY)
+        val value = mock<Span>()
+
+        // When
+        val result = testedContext.with(key, value)
+
+        // Then
+        assertThat(result).isInstanceOf(OtelContext::class.java)
```

Comment:
description of `with` [says](https://github.com/open-telemetry/opentelemetry-java/blob/001f34f1703a43cdd473d2a94d15c125f6d2d622/context/src/main/java/io/opentelemetry/context/Context.java#L149):

> Returns a new context with the given key value set.

So should we check also that `result` contains the `value`? Here and in the similar tests below.

Reply:
yup it makes sense, good point.